### PR TITLE
fix: Fixing git-urls dependency using another fork of the repo

### PR DESCRIPTION
**Pull request by BKirov**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [x] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [x] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Comment by crenshaw-dev** [comment]
Thanks, @BKirov! Can you fix the DCO check? Instructions are in the details link of the failed check.

**Comment by BKirov** [comment]
Fixed @crenshaw-dev

File: go.mod [hunk 1]
```diff
@@ -138,6 +138,7 @@ require (
 	github.com/tidwall/gjson v1.14.4 // indirect
 	github.com/tidwall/match v1.1.1 // indirect
 	github.com/tidwall/pretty v1.2.0 // indirect
+	github.com/whilp/git-urls v0.0.0-20191001220047-6db9661140c0 // indirect
```
Comment (jannfis) [inline go.mod:141]: I was wondering if we should use a `replace` rule to get rid of all the instances in `github.com/whilp/git-urls` across our dependencies.

Comment (crenshaw-dev) [reply go.mod:141]: Yeah good idea, come to think of it @BKirov that's the only way to be sure we clear up your image scanners.

**Comment by BKirov** [comment]
@crenshaw-dev  @jannfis  can you please approve

File: go.mod (hunk 1 above)
Comment (BKirov) [reply go.mod:141]: fixed , can you please approve

Comment (crenshaw-dev) [reply go.mod:141]: Hm. The vunlerable version is still showing up as an indirect dependency. Do you need two replaces?

```
github.com/whilp/git-urls v0.0.0-20191001220047-6db9661140c0 // indirect
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

**Comment by BKirov** [comment]
@crenshaw-dev  can you help with that ? where to put the second replace ? this line for the old github.com/whilp/git-urls v0.0.0-20191001220047-6db9661140c0 // indirect  
was automatically created ....

**Comment by crenshaw-dev** [comment]
@BKirov I'm not super up on how go mod works, but I think you could just add it after the existing replace.

> was automatically created

Yep, because it's a transient dependency. So we haven't quite eliminated the vulnerable code yet.

**Comment by BKirov** [comment]
SO i am addint it like that ? 
	// Avoid  CVE-2023-46402
	github.com/whilp/git-urls v1.0.0 => github.com/chainguard-dev/git-urls v1.0.2
	github.com/whilp/git-urls v0.0.0-20191001220047-6db9661140c0 => github.com/chainguard-dev/git-urls v1.0.2
	
	`bkirov@OT-MAC-C02GX0AZQ6LX argo-cd % go mod tidy
go: github.com/chainguard-dev/git-urls@v1.0.2 used for two different module paths (github.com/chainguard-dev/git-urls and github.com/whilp/git-urls)`

File: go.mod [hunk 2]
```diff
@@ -298,6 +299,9 @@ replace (
 	github.com/golang/protobuf => github.com/golang/protobuf v1.4.2
 	github.com/grpc-ecosystem/grpc-gateway => github.com/grpc-ecosystem/grpc-gateway v1.16.0
 
+	// Avoid  CVE-2023-46402
+	github.com/whilp/git-urls v1.0.0 => github.com/chainguard-dev/git-urls v1.0.2
```
Comment (crenshaw-dev) [inline go.mod:303]:
```suggestion
	github.com/whilp/git-urls => github.com/chainguard-dev/git-urls v1.0.2
```

Ah. That oughta do it.

Comment (crenshaw-dev) [reply go.mod:303]: Don't commit the suggestion here though, do it locally and then run go mod tidy.

**Comment by BKirov** [comment]
@crenshaw-dev  I am getting this locally and then here too : 

go: github.com/chainguard-dev/git-urls@v1.0.2 used for two different module paths (github.com/chainguard-dev/git-urls and github.com/whilp/git-urls)

**Comment by crenshaw-dev** [comment]
Trying fresh here: https://github.com/argoproj/argo-cd/pull/17732

**Comment by crenshaw-dev** [comment]
Merged here: https://github.com/argoproj/argo-cd/pull/17732

Thanks for your patience @BKirov, it turned out to be trickier than expected. :-)

**Pull request closed without merging**